File: framework/Button.js
const BaseElement = require('./BaseElement.js')
const logger = require('./logger.js')

class Button extends BaseElement {
  constructor(selector, name) {
    super(selector, name)
  }

  async isEnabled() {
    logger.debug(`Checking if ${this.name} is enabled`)    
    const isEnabled = await (await this._getElement()).isEnabled()
    logger.debug(`${this.name} enabled: ${isEnabled}.`)    
    return isEnabled
  }
}

module.exports = Button

File: src/test/java/framework/elements/Button.java
package framework.elements;

import org.openqa.selenium.*;
import framework.utils.LoggerUtil;

public class Button extends BaseElement {

    public Button(By locator, String name){
        super(locator, name);
    }

    public boolean IsEnabled(){
        LoggerUtil.debug("Checking if {} is enabled", name);
        var isEnabled = _GetElement().isEnabled();
        LoggerUtil.debug("Element {} enabled: {}", name, isEnabled);
        return isEnabled;
    }

}